CriticalMass/AppDataStore.swift

```diff
@@ -22,15 +22,8 @@ public class AppDataStore: DataStore {
     private var lastKnownResponse: ApiResponse?
 
     public var userName: String {
-        set {
-            if !newValue.isEmpty {
-                userDefaults.set(newValue, forKey: #function)
-            }
-        }
-
-        get {
-            userDefaults.string(forKey: #function) ?? UIDevice.current.name
-        }
+        get { userDefaults.username ?? UIDevice.current.name }
+        set { userDefaults.username = newValue }
```

Comment:
We need to check if the new name isn't an empty string. But we can also move that check to the UI

Reply:
Think it is better located there since we would want to react to it, right?

Reply:
Just had a look and I am not sure where to actually change it in the UI. `FriendSettingsTableViewCell`?

Reply:
We can just ignore the input if it's empty and not pass it to the DataStore and show the old input again. That is also the behaviour of the current implementation

Reply:
So where to do that? I honestly don’t know where to adjust